packages/@uppy/core/src/Uppy.js

```diff
@@ -261,6 +232,8 @@ class Uppy {
 
     this.i18nInit()
 
+    this.#restricter.opts = this.opts
```

Comment:
Instead of reassigning this all the time, maybe in the Restrictors constructor we could instead pass a function `getOptions: () => this.opts` that it can call. that way we don't have to duplicate the state in two places?